src/main/java/net/coderbot/iris/shaderpack/ShaderProperties.java

```diff
@@ -192,8 +198,44 @@ public ShaderProperties(String contents, ShaderPackOptions shaderPackOptions, It
 
 			handlePassDirective("blend.", key, value, pass -> {
 				if (pass.contains(".")) {
-					// TODO: Support per-buffer blending directives (glBlendFuncSeparateI)
-					Iris.logger.warn("Per-buffer pass blending directives are not supported, ignoring blend directive for " + key);
+
+					if (!IrisRenderSystem.supportsBufferBlending()) {
+						throw new RuntimeException("Buffer blending is not supported on this platform, however it was attempted to be used!");
+					}
+
+					String[] parts = pass.split("\\.");
+					int index = PackRenderTargetDirectives.LEGACY_RENDER_TARGETS.indexOf(parts[1]);
+
+					if (index == -1 && parts[1].startsWith("colortex")) {
+						String id = parts[1].substring("colortex".length());
+
+						try {
+							index = Integer.parseInt(id);
+						} catch (NumberFormatException e) {
+							throw new RuntimeException("Failed to parse buffer blend!", e);
+						}
+					}
+
+					if (index == -1) {
+						throw new RuntimeException("Failed to parse buffer blend! index = " + index);
+					}
+
+					if ("off".equals(value)) {
+						bufferBlendOverrides.computeIfAbsent(parts[0], list -> new ArrayList<>()).add(new BufferBlendOverride(index, null));
+						return;
+					}
+
+					String[] modeArray = value.split(" ");
+					int[] modes = new int[4];
+
+					int i = 0;
+					for (String modeName : modeArray) {
+						modes[i] = BlendModeFunction.fromString(modeName).get().getGlId();
```

Comment:
We could get an `ArrayIndexOutOfBounds` Exception here due to the 229th line.
Shouldn't that be?
`int[] modes = new int[modeArray.length];`

Reply:
Thanks for pointing this out, fixed